modules/standard/Map.chpl

```diff
@@ -295,8 +311,16 @@ module Map {
       :yields: A reference to one of the values contained in this map.
     */
     iter values() ref {
-      for val in vals {
-        yield val;
+      if chpl_nilable {
+        for val in vals {
+          yield val;
+        }
+      } else {
+        try! {
+          for val in vals {
+            yield val: valType;
```

Comment:
Are we sure that this doesn't just return a reference to a temporary value? In my limited experience it didn't seem to be possible to return a reference to a nilable cast to non-nilable (as the cast created a new value).

Reply:
*Sigh*. Yes, you're right. I need to figure out a different way.

Reply:
My thought is that for iterators, we should only support only overloads that return `const` or not support them at all, for this release. The same would go for `this`, of the available overloads, only the one that returns `const` would work (using where clauses) for non-nilable classes.

Reply:
I also think it is reasonable to want the ability to do an unsafe cast of a reference from nil -> non-nil, which is discussed here: https://github.com/chapel-lang/chapel/issues/14937